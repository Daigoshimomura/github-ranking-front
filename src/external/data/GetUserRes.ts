export interface GetUserRes {
  user: User;
}
export interface User {
  userId: String;
  avatarUrl: String;
  repositories: Repositories;
  followers: Followers;
  pullRequests: PullRequests;
  issues: Issues;
}

interface Repositories {
  nodes: Node[];
  totalCount: Number;
}

interface Node {
  id: String;
  name: String;
  url: String;
  primaryLanguage: Language;
  forks: Forks;
  watchers: Wachers;
  viewerHasStarred: false;
  stargazers: Stargazers;
}

interface Language {
  name: String;
}
interface Forks {
  totalCount: 0;
}
interface Wachers {
  totalCount: 0;
}
interface Stargazers {
  totalCount: 0;
}

interface Followers {
  totalCount: Number;
}

interface PullRequests {
  totalCount: Number;
}

interface Issues {
  totalCount: Number;
}
