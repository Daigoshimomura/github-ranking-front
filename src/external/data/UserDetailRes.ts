export interface UserDetailRes {
  userName: string;
  userId: string;
  avatarUrl: string;
  githubUrl: string;
  tier: string;
  rank: Number;
  score: Number;
  currentNumber: Number;
  followersCount: Number;
  issuesCount: Number;
  pullRequestCount: Number;
  repositoriesCount: Number;
  forksCountTotal: Number;
  stargazerCountTotal: Number;
  watchersCountTotal: Number;
  mainLanguage: string;
  lastupdateDate: Date;
}
