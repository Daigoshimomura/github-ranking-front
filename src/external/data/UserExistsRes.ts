export interface UserExistsRes {
  exists: boolean;
}
