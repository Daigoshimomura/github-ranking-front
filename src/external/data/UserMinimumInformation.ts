export interface UserMinimumInformation {
  userId: string;
  score: Number;
  mainLanguage: string;
  avatarUrl: string;
}
