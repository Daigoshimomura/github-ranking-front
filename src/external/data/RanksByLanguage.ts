import { UserMinimumInformation } from "./UserMinimumInformation";

export interface RanksByLanguage {
  language: String;
  userInfomations: UserMinimumInformation[];
}
