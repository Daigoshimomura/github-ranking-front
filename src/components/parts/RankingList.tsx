import React from "react";
import { RanksByLanguage } from "../../external/data/RanksByLanguage";
import { Typography, Grid, List } from "@material-ui/core";
import { RankingListItem } from "./RankingListItem";

interface RankingListProps {
  ranksByLanguage: RanksByLanguage;
  itemClickCallback: Function;
}

export class RankingList extends React.Component<RankingListProps, {}> {
  render() {
    return (
      <Grid item xs={12} md={4}>
        <Typography variant="h6">
          {this.props.ranksByLanguage.language}
        </Typography>
        <div>
          <List>
            {this.props.ranksByLanguage.userInfomations.map((t, index) => (
              <RankingListItem
                key={t.userId}
                user={t}
                rank={index + 1}
                itemClickCallback={this.props.itemClickCallback}
              />
            ))}
          </List>
        </div>
      </Grid>
    );
  }
}
