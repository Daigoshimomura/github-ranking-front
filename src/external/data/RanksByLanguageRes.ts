import { RanksByLanguage } from "./RanksByLanguage";

export interface RankgByLanguageRes {
  rankByLanguages: RanksByLanguage[];
}
