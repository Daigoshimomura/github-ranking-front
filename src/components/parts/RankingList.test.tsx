import { mount } from "enzyme";
import React from "react";
import { RankingList } from "./RankingList";
import { RanksByLanguage } from "../../external/data/RanksByLanguage";
import { Grid } from "@material-ui/core";
import renderer from "react-test-renderer";

const ranksByLanguage: RanksByLanguage = {
  language: "language",
  userInfomations: []
};
const testMock = jest.fn();

describe("RankingList", () => {
  it("snapshot", () => {
    const tree = renderer
      .create(
        <RankingList
          ranksByLanguage={ranksByLanguage}
          itemClickCallback={testMock}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("レンダリングの確認", () => {
    const wrapper = mount(
      <RankingList
        ranksByLanguage={ranksByLanguage}
        itemClickCallback={testMock}
      />
    );
    expect(wrapper.find(Grid).length).toBe(1);
  });
  it("propsをちゃんと渡せてることの確認", () => {
    const wrapper = mount(
      <RankingList
        ranksByLanguage={ranksByLanguage}
        itemClickCallback={testMock}
      />
    );
    expect(wrapper.prop("ranksByLanguage").language).toBe("language");
    expect(wrapper.prop("itemClickCallback")).toBe(testMock);
  });
});
