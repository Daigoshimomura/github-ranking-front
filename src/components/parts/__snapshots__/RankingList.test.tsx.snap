// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RankingList snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
>
  <h6
    className="MuiTypography-root MuiTypography-h6"
  >
    language
  </h6>
  <div>
    <ul
      className="MuiList-root MuiList-padding"
    />
  </div>
</div>
`;
